sdk/storage/azure-storage-file-datalake/azure/storage/file/out/datalake/_shared_access_signature.pyi

```diff
@@ -0,0 +1,10 @@
+# Stubs for datalake._shared_access_signature (Python 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from typing import Any, Optional
+
+def generate_account_sas(account_name: str, account_key: str, resource_types: Union[ResourceTypes, str], permission: Union[AccountSasPermissions, str], expiry: Optional[Union[datetime, str]], start: Optional[Union[datetime, str]]=..., ip: Optional[str]=..., **kwargs: Any) -> str: ...
+def generate_file_system_sas(account_name: str, file_system_name: str, account_key: Optional[str]=..., user_delegation_key: Optional[UserDelegationKey]=..., permission: Optional[Union[FileSystemSasPermissions, str]]=..., expiry: Optional[Union[datetime, str]]=..., start: Optional[Union[datetime, str]]=..., ip: Optional[str]=..., **kwargs: Any) -> str: ...
+def generate_directory_sas(account_name: str, file_system_name: str, directory_name: str, account_key: Optional[str]=..., user_delegation_key: Optional[UserDelegationKey]=..., permission: Optional[Union[BlobSasPermissions, str]]=..., expiry: Optional[Union[datetime, str]]=..., start: Optional[Union[datetime, str]]=..., ip: Optional[str]=..., **kwargs: Any) -> str: ...
+def generate_file_sas(account_name: str, file_system_name: str, directory_name: str, file_name: str, account_key: Optional[str]=..., user_delegation_key: Optional[UserDelegationKey]=..., permission: Optional[Union[BlobSasPermissions, str]]=..., expiry: Optional[Union[datetime, str]]=..., start: Optional[Union[datetime, str]]=..., ip: Optional[str]=..., **kwargs: Any) -> str: ...
```

Comment:
should we have both generate_file_sas and generate_file_system_sas? if we want to keep both, do we want to make directory name optional(the same question as DataLakeFileClient Constructor)